web/src/pages/Main.tsx

```diff
@@ -1,14 +1,65 @@
 import { Header } from '../components/common/Header'
-import { useContext } from 'react'
-import { AuthContext } from 'react-oauth2-code-pkce'
+import { useEffect, useState } from 'react'
+import { MoleTable } from '../components/feature/MoleTable'
+import MercuryAPI from '../api/MercuryAPI'
+import { AxiosResponse } from 'axios'
+
+// TODO: Get type from generated API
+export type TComponentResponse = {
+  [key: string]: { [key: string]: string }
+}
 
 export const MainPage = (): JSX.Element => {
-  const { token } = useContext(AuthContext)
+  const [components, setComponents] = useState<TComponentResponse>()
+  const [loading, setLoading] = useState<boolean>(false)
+
+  // Fetch list of components name once on page load
+  useEffect(() => {
+    const mercuryApi = new MercuryAPI('')
+    mercuryApi
+      .getComponents()
+      .then((response: AxiosResponse<TComponentResponse>) =>
+        setComponents(response.data)
+      )
+      .finally(() => setLoading(false))
+  }, [])
+
+  // TODO: Remove with actual calculation results
+  const multiflashResult = {
+    phase_values: {
```

Comment:
Snake_case 📘

Reply:
the temp mock data represent the values returned from the API